net/libreswan/files/etc/hotplug.d/libreswan/02-vti

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+. /lib/functions.sh
+
+[ "${PLUTO_VERB}" != "route-client" ] && [ "${PLUTO_VERB}" != "up-client" ] && exit 0
+
+CONNECTION=${PLUTO_CONNECTION%/*}
+[ -z "$CONNECTION" ] && exit 0
+
+update_peeraddr=$(uci_get libreswan $CONNECTION update_peeraddr)
+[ "$update_peeraddr" != "1" ] && exit 0
+
+interface=$(uci_get libreswan $CONNECTION interface)
+[ -z "$interface" ] && exit 0
+
+proto=$(uci_get network "$interface" proto)
+[ "$proto" != "vti" ] && exit 0
+
+peeraddr=$(uci_get network "$interface" peeraddr)
+[ "$peeraddr" == "$PLUTO_PEER" ] && exit 0
+
+uci_set network "$interface" peeraddr "$PLUTO_PEER"
+uci_commit network
```

Comment:
I think that's a bit daring. Every time libreswan is executing the hotplug, the network configuration is written here. If this is really necessary, I would prefer to work with the uci -p option and then move it to /var/state.
See https://github.com/openwrt/openwrt/blob/master/package/system/uci/files/lib/config/uci.sh#L63
and
See https://github.com/openwrt/openwrt/blob/master/package/system/uci/files/lib/config/uci.sh#L55

Reply:
It is actually not all the time, but instead, it does this only when configured and required.

It depends on the uci option `update_peeraddr`, It also exits if `peeraddr` is the same as before.

Reply:
Yes, I am aware of  `uci_set_state` but here `peeraddr` is of VTI interface from `/etc/config/network`. VTI interface has the limitation that prevents multiple VTI tunnels with `peeraddr` of `0.0.0.0/0`, If the remote endpoint is using dynamic NAT and there is a need to create multiple VTI interfaces, this option would auto-update VTI interface `peeraddr` based on `PLUTO_PEER` and it only gets updated if `update_peeraddr` is enabled and `peeraddr` is not same as `PLUTO_PEER`